filebeat/input/file.go

```diff
@@ -72,7 +77,37 @@ func mergeJSONFields(f *FileEvent, event common.MapStr) {
 	if f.JSONConfig.KeysUnderRoot {
 		for k, v := range f.JSONFields {
 			if f.JSONConfig.OverwriteKeys {
-				event[k] = v
+				if k == "@timestamp" {
+					vstr, ok := v.(string)
+					if !ok {
+						logp.Err("JSON: Won't overwrite @timestamp because value is not string")
+						event[jsonErrorKey] = "@timestamp not overwritten (not string)"
+						continue
+					}
+					// @timestamp must be of time common.Time
+					ts, err := common.ParseTime(vstr)
```

Comment:
Same as above, didn't meant for this to be super configurable for now, to force people to use things like Logstash and Elasticsearch ingest node to parsing and such. The main goal was to make sure we don't panic, which was the case in the bug report.